### PR TITLE
Mysql change xcom value col type for MySQL backend

### PR DESCRIPTION
**Pull request by pankajastro**
Xcom value we can store Airflow metadata is a bit smaller
when we use MySQL as a database backend since by default,
the Sqlalchemy map `LargeBinary` to MySQL blob type 
which can store up to 65,535 bytes. In this PR,
 I'm proposing to change the `value` column type to 
longblob for xcom table if a user using MySQL database backend
https://dev.mysql.com/doc/refman/8.0/en/string-type-syntax.html

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/migrations/versions/0138_2_9_0_make_xcom_value_to_longblob_for_mysql.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+down_revision = '8e1c784a4fc7'
+branch_labels = None
+depends_on = None
+airflow_version = '2.9.0'
+
+
+def upgrade():
+    """Apply Change value column type to longblob in xcom table for mysql"""
+    conn = op.get_bind()
+    if conn.dialect.name == "mssql":
```
Comment (ephraimbuddy) [inline airflow/migrations/versions/0138_2_9_0_make_xcom_value_to_longblob_for_mysql.py]: You mean `mysql` here?

Comment (pankajastro) [reply airflow/migrations/versions/0138_2_9_0_make_xcom_value_to_longblob_for_mysql.py]: yes, thanks for pointing

**Comment by potiuk** [comment]
Will that cause a long migration process possibly ? or will such migration be rather fast ? I suspect if there is something to rewrite the way how the blobs are stored, it might take an awfully long time to run such a migration. 

I think, if that's the case then we will likely have to at least warn the users in significant note that it might happen and that they should likely run `db clean` on xcom table for some old data?

**Comment by pankajastro** [comment]
> Will that cause a long migration process possibly ?

Yes, that will be the case if they have lots of large data in Xcom, although I don't have a specific number at the moment.

> I think, if that's the case then we will likely have to at least warn the users in significant note that it might happen and that they should likely run db clean on xcom table for some old data?

This makes sense. I'll add a warning. 
If the blob size > 65,535 bytes then to downgrade they will have to clean at least large data.

**Comment by pankajastro** [comment]
@potiuk I have tried to add a note  `docs/apache-airflow/howto/set-up-database.rst` but not sure if this right place. any suggestions?

File: docs/apache-airflow/howto/set-up-database.rst [hunk 2]
```diff
@@ -318,6 +318,13 @@ In addition, you also should pay particular attention to MySQL's encoding. Altho
     Read https://stackoverflow.com/questions/9192027/invalid-default-value-for-create-date-timestamp-field for how to disable it.
     See `SQL Mode - NO_ZERO_DATE <https://dev.mysql.com/doc/refman/8.0/en/sql-mode.html#sqlmode_no_zero_date>`__ for more information.
 
+.. warning::
+  In Airflow 2.9.0 we have changed xcom table column value type from blob to longblob. This will allow you to store relatively big data in Xcom but process can take a significant amount of time if you have a lot of large data stored in Xcom.
```
Comment (jedcunningham) [inline docs/apache-airflow/howto/set-up-database.rst]: This should go in a newsfragment instead -  [docs here](https://github.com/apache/airflow/blob/83d62cae695ebea0e537c7bfa8e9cc65de05ec09/contributing-docs/16_contribution_workflow.rst#step-4-prepare-pr).

**Comment by potiuk** [comment]
Should be a significant newsfragment too.

**Review by jedcunningham — COMMENTED** [review]
We should definitely test whether this will rewrite all of xcom - it might not.

If it does, and it results in a slow migration, I really wonder if we should _not_ do this and instruct users to set up an object storage xcom backend instead. Thoughts?

**Comment by Taragolis** [comment]
> I think, if that's the case then we will likely have to at least warn the users in significant note that it might happen and that they should likely run db clean on xcom table for some old data?

A bit Friday jokes: If take in account [last survey](https://docs.google.com/forms/d/1wYm6c5Gn379zkg7zD7vcWB-1fCjnOocT0oZm-tjft_Q/viewanalytics) - number of MySQL users about 11-14 % (11% MySQL 8, and 3.4% MySQL 5) for avoid long migration it could be an option to start from scratch with Postgres 🤣 .

**Review by Taragolis — APPROVED** [review]

**Comment by potiuk** [comment]
> > I think, if that's the case then we will likely have to at least warn the users in significant note that it might happen and that they should likely run db clean on xcom table for some old data?
> 
> A bit Friday jokes: If take in account [last survey](https://docs.google.com/forms/d/1wYm6c5Gn379zkg7zD7vcWB-1fCjnOocT0oZm-tjft_Q/viewanalytics) - number of MySQL users about 11-14 % (11% MySQL 8, and 3.4% MySQL 5) for avoid long migration it could be an option to start from scratch with Postgres 🤣 .

Yes. Let's also add "if mysql: sleep(0.2)" in various places in Airlfow. Enough to be impacting performance, but not enough to say "MYSQL does not work". Then we will be able to tell "But Postgres is WAY faster".

**Comment by kaxil** [comment]
>If it does, and it results in a slow migration, I really wonder if we should not do this and instruct users to set up an object storage xcom backend instead. Thoughts?

The current size is low per row - 64kb for MySQL, so increasing to 64 MB should be supported without Custom XCom backend

**Review by kaxil — APPROVED** [review]

**Review by dirrao — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**